railgun/apis/addtoscheme_knative_v1alpha1.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2021 Kong, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package apis
+
+import (
+	"knative.dev/networking/pkg/apis/networking/v1alpha1"
+)
+
+func init() {
+	// Register the types with the Scheme so the components can map objects to GroupVersionKinds and back
+	AddToSchemes = append(AddToSchemes, v1alpha1.SchemeBuilder.AddToScheme)
```

Comment:
It doesn't seem like we would need to add add knative.Ingress to the runtime.Scheme on package load, but instead the controller manager would need to add it prior to starting the controller if it's enabled. I would suggest we at least look into an alternative approach here where we only add knative to the scheme opportunistically if its enabled.

Reply:
Would like to follow good strategy - apply runtime scheme opportunistically.

Reply:
In that case are you on board with dropping these init() scheme loaders in favor of loading only when enabled in the manager?

Reply:
Answer your question in another similar thread.